### PR TITLE
YD-136 Set UserDTO to null when user removed

### PR DESCRIPTION
**Pull request by aristotelos**
Not returning a 'stub entity'

### CODE REVIEW

File: core/src/main/java/nu/yona/server/subscriptions/service/UserDTO.java [hunk 1]
```diff
@@ -154,17 +154,12 @@ static UserDTO createInstance(User userEntity)
 	{
 		if (userEntity == null)
 		{
-			return createRemovedUserInstance();
+			return null;
```
Comment (Bert-R) [inline core/src/main/java/nu/yona/server/subscriptions/service/UserDTO.java]: Can't you fix this at the level of the caller? It's a weird contract to pass a null value to a ``createInstance`` method and then get null back.

File: core/src/main/java/nu/yona/server/subscriptions/service/BuddyDisconnectMessageDTO.java [hunk 2]
```diff
@@ -61,10 +61,9 @@ public boolean isProcessed()
 	public static BuddyDisconnectMessageDTO createInstance(UserDTO actingUser, BuddyDisconnectMessage messageEntity)
 	{
 		User userEntity = messageEntity.getUser(); // may be null if deleted
-		UserDTO user = userEntity != null ? UserDTO.createInstance(userEntity) : UserDTO.createRemovedUserInstance();
```
Comment (Bert-R) [inline core/src/main/java/nu/yona/server/subscriptions/service/BuddyDisconnectMessageDTO.java]: If you just retain the below line, the API contract for ``createInstance`` can be regular:
```
UserDTO user = userEntity == null ? null : UserDTO.createInstance(userEntity);
```

**Comment by Bert-R** [comment]
Comments added.

File: core/src/main/java/nu/yona/server/subscriptions/service/UserDTO.java (hunk 1 above)
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/subscriptions/service/UserDTO.java]: There were 4 callers that needed the `null` check. I have created an alternative method `createInstanceIfNotNull` that returns `null` or an instance and added some comments to the methods to explain it.

Comment (Bert-R) [inline core/src/main/java/nu/yona/server/subscriptions/service/UserDTO.java]: Yes, there are more of them. I didn't notice.

**Pull request merged**